WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingHomeItemUiState.kt

```diff
@@ -16,28 +16,40 @@ sealed class PrepublishingHomeItemUiState {
     data class HeaderUiState(val siteName: UiStringText, val siteIconUrl: String) :
             PrepublishingHomeItemUiState()
 
-    sealed class SubmitButtonUiState(
+    sealed class ButtonUiState(
         val buttonText: UiStringRes,
         val publishPost: PublishPost,
         open val onButtonClicked: (PublishPost) -> Unit
```

Comment:
I believe we can move this away from the constructor - it can be a regular field (in ButtonUiState). We won't need to pass it as an argument in all the children. The `override` keyword will make sure to write the value in to the correct field. Wdyt? (similar [example](https://github.com/wordpress-mobile/WordPress-Android/blob/9bdfb35a3cd085544a21b8f05d9f35ea59518861/WordPress/src/main/java/org/wordpress/android/ui/reader/subfilter/SubfilterListItem.kt#L17))

Reply:
Nice. Thanks for this 😉

Reply:
cdcc032